modules/auxiliary/gather/hivenightmare_windows_sam_leak.rb

```diff
@@ -0,0 +1,82 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+    Rank = ExcellentRanking
+  
+    include Msf::Post::File
+    prepend Msf::Exploit::Remote::AutoCheck
+    include Msf::PostMixin
+  
+    def initialize(info = {})
+      super(
+        update_info(
+          info,
+          'Name' => 'HiveNighmare - SAM Leak',
+          'Description' => %q{
+              to do
+          },
+          'License' => MSF_LICENSE,
+          'Author' =>
+            [       
+              'romarroca',       # POC on www.github.com
+              'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+            ],
+          'References' =>
+            [
+              ['CVE', '2021-36934'],
+              ['URL', 'https://github.com/romarroca/SeriousSam']
```

Comment:
PoC exploit:

* https://github.com/GossiTheDog/HiveNightmare

Associated writeup:

* https://doublepulsar.com/hivenightmare-aka-serioussam-anybody-can-read-the-registry-in-windows-10-7a871c465fa5

There are a bunch of other posts about the bug. ISC has one of the better writeups:

* https://isc.sans.edu/diary/Summer+of+SAM+-+incorrect+permissions+on+Windows+1011+hives/27652